cmd/metal-api/internal/datastore/integer.go

```diff
@@ -51,72 +68,79 @@ type Integerinfo struct {
 // - releasing the integer is fast
 // - you do not have gaps (because you can give the integers back to the pool)
 // - everything can be done atomically, so there are no race conditions
-func (rs *RethinkStore) initIntegerPool() error {
-
-	var result Integerinfo
-	err := rs.findEntityByID(rs.integerInfoTable(), &result, "integerpool")
+func (rs *RethinkStore) initIntegerPool(tablename string, min, max uint) (*IntegerPool, error) {
+	var result integerinfo
+	err := rs.findEntityByID(rs.integerInfoTable(tablename), &result, tablename)
 	if err != nil {
 		if !metal.IsNotFound(err) {
-			return err
+			return nil, err
 		}
 	}
 
+	ip := &IntegerPool{
+		tablename: tablename,
+		min:       min,
+		max:       max,
+		rs:        rs,
```

Comment:
Do we need the `rs`? Isn't `tablename` enough for all actions after construction?

Reply:
rs is required to actually read and write to the database:
https://github.com/metal-stack/metal-api/pull/108/files#diff-d354d7679d95da26fd3b929143a9a955R104

Reply:
store term and session instead of rs in integerpool